Adding a justfile, cross platform build system, UI updates

tristanpoland opened the pull request:
This pull request introduces several improvements to cross-platform script execution and developer tooling, mainly by replacing fragile shell-based workarounds with a more robust Node.js wrapper, and by adding a new, user-friendly `justfile` build system. It also enhances diagnostics for the entity catalog in the Angular app and introduces a dedicated script for building custom Angular builders.

**Cross-platform script execution improvements:**
- Introduced a new Node.js script runner `build/run-script-cjs.cjs` to replace previous shell-based renaming and execution of `.js`/`.cjs` scripts, ensuring compatibility across platforms and simplifying script invocation in `package.json`, `angular.json`, and `build/post-setup.cjs`. [[1]](diffhunk://#diff-717aac61e610c8fe90596230969d3d82c510aaf9f9d3ed19375ba5b1732e526bR1-R61) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L77-R81) [[3]](diffhunk://#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7dL18-R18) [[4]](diffhunk://#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7dL27-R27) [[5]](diffhunk://#diff-4792073fe2d24e17edd29891cd30ec0dbe206b830ce03f5f2f0774a046d51885L24-R26)
- Updated all relevant npm scripts and Angular prebuild scripts to use the new runner for scripts like `clean-symlinks`, `store-git-metadata`, `dev-setup`, and `prepare-backend`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L77-R81) [[2]](diffhunk://#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7dL18-R18) [[3]](diffhunk://#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7dL27-R27)

**Developer tooling and build system:**
- Added a new `justfile` providing a cross-platform, PowerShell-friendly build and development workflow with clear targets for setup, building, testing, security, release management, and cleanup. This enhances onboarding and daily developer experience, especially on Windows.
- Replaced the shell-based `build-builders` npm script with a new cross-platform Node.js script `build/build-builders.cjs` to build and copy custom Angular builders, improving reliability and maintainability. [[1]](diffhunk://#diff-5bd7344c604fc57230321a9f322c5fae7a787694663477975aa503ec44af3a2fR1-R54) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L24-R24)

**Debugging and diagnostics:**
- Added detailed entity catalog diagnostics to the Angular app's `AppComponent` initialization to help debug issues with fresh clones or entity registration, including logging of registered endpoints, entities, and endpoint types. [[1]](diffhunk://#diff-c37ffd7c73e94fcd95bef1fd9ca490a5429e02bc38b931940e09798d18e005d3L4-R4) [[2]](diffhunk://#diff-c37ffd7c73e94fcd95bef1fd9ca490a5429e02bc38b931940e09798d18e005d3R74-R90)


## Note

This should not break the `make` build system in any way. `just` was added because it has better cross platform performance consistency.

tristanpoland committed:
Add git routing module and update public API

Introduced git-routing.module.ts to the git package and updated public_api.ts to export the new module. This enables routing functionality for the git package.

tristanpoland committed:
Refactor build scripts to ES module syntax

Converted build scripts from CommonJS to ES module syntax, updating imports/exports and direct execution checks. Also added 'routingModule' to the git package.json for improved module configuration.

tristanpoland committed:
Update package.json

tristanpoland committed:
Add args support for build tools in orchestrator

Introduces the ability to specify command-line arguments for build tools in the orchestrator. Updates SASS Resolver and Index Transformer tool definitions to include relevant args and modifies tool execution logic to pass these arguments.

tristanpoland committed:
Add custom builders build step to post-setup script

Introduces a function to build custom Angular builders during post-install setup. Also updates .gitignore to exclude the builders' dist directory.

tristanpoland committed:
Add diagnostics logging for entity catalog and endpoint modal

Introduces debug logging in AppComponent and EndpointRegisterModalComponent to help diagnose issues with entity catalog and endpoint registration tiles. Also updates index.html to use static metadata values and replaces template placeholders with actual values.

tristanpoland committed:
Update build-orchestrator.js

tristanpoland committed:
Update build-orchestrator.js

tristanpoland committed:
Merge branch 'angular20/experimental' of https://github.com/tristanpoland/stratos into angular20/experimental

tristanpoland committed:
Add cross-platform script runner and update build scripts

Introduced build/run-script-cjs.cjs for cross-platform execution of scripts requiring .cjs extension. Updated package.json, angular.json, and build/post-setup.cjs to use this runner, replacing previous shell-based renaming logic. Added build/build-builders.cjs for building Angular builders in a cross-platform way. Added a justfile for unified build/dev commands. Improved devkit build to ensure package.json uses commonjs type. These changes improve Windows compatibility and streamline build tooling.

tristanpoland committed:
Switch to Material Symbols and update card styles

Replaces usage of 'material-icons' with 'material-symbols-outlined' throughout the frontend for icon consistency. Updates card content padding from p-4 to p-6 in multiple components and standardizes card header titles to use <h3> instead of <h2> or <div>. Also simplifies some button and layout classes for improved UI consistency.

tristanpoland committed:
Add Node.js backend build scripts and default config

Introduced build-backend.cjs and build-backend.js scripts to handle backend Go binary builds in a cross-platform way. Updated the justfile to use the new Node.js build script for backend builds. Added a default developer configuration file at src/jetstream/.config for backend development.

tristanpoland committed:
Redesign login page with glassmorphism and split layout

Revamps the login page UI to use a modern split-screen layout with glassmorphism effects, including a branding/feature panel and a refined login form. Adds support for a 'Remember me' option, feature highlights, and theme-driven customization for gradients and glass parameters. Updates related SCSS, theme config, and login logic to support new design and features.

tristanpoland committed:
Add smooth login page transition animations

Introduced Angular route and login transition animations for the login page, including a new route-animations.ts file. Updated the login page component to use the new animation, enhanced SCSS for smoother transitions, and improved navigation fallback logic to wait for animation completion before hard redirects. Also added a background overlay in main.scss to prevent white flashes during route transitions.

norman-abramovitz commented:
@tristanpoland I am wondering if you can help resolve the dependency failures listed above?

tristanpoland commented:
> @tristanpoland I am wondering if you can help resolve the dependency failures listed above?

To which are you referring? I'm not sure I know what you mean.

tristanpoland committed:
Refine UI styles and layout for modernized look

This update polishes the frontend UI with improved spacing, rounded corners, enhanced button and card styles, smoother transitions, and subtle animations. It affects page headers, home and endpoints pages, custom cards, and global theme styles, providing a more modern, consistent, and visually appealing user experience. Also adds a default backend developer configuration file for local development.

tristanpoland committed:
Refine UI styles for shared components

Improves the visual appearance and interactivity of breadcrumbs, chips, dialog-confirm, loading-page, and no-content-message components. Updates include enhanced transitions, spacing, hover/active effects, modernized sizing, and improved accessibility and theme support for a more polished user experience.

tristanpoland committed:
Refine dashboard and endpoint form UI/UX styles

Improves the dashboard side navigation and help panel transitions, adds browser-specific animation handling, and enhances focus/hover states for better accessibility. The edit endpoint form is restructured for clarity, with improved sectioning, labels, and input styling, including a new SSO code block with copy button and more consistent advanced/SSL configuration sections. SCSS updates provide a more modern, accessible, and visually consistent user experience across dashboard and endpoint editing components.

norman-abramovitz commented:
The comment below showing check failures is showing merge conflicts.   I am wondering if you can fix these.  We can do it together as well
@tristanpoland 

<img width="822" height="263" alt="image" src="https://github.com/user-attachments/assets/a333fa8b-87c0-4344-aad2-7a2f19f1a4e7" />
